src/instructlab/model/torchrun_train.py

```diff
@@ -0,0 +1,304 @@
+    tokenizer_utils,
+    utils,
+)
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import Adafactor, AutoModelForCausalLM
+import numpy as np
+import psutil
+import torch
+import torch.nn.functional as F
+
+# First Party
+from instructlab.llamacpp import llamacpp_convert_to_gguf
+
+
+def main(train_args):
+    dp.main(
+        config.DataProcessArgs(
+            data_output_path=train_args.data_output_dir,
+            model_path=train_args.model_path,
+            data_path=train_args.data_path,
+            max_seq_len=train_args.max_seq_len,
+            chat_tmpl_path=train_args.chat_tmpl_path,
+        )
+    )
+    CHAT_TEMPLATE, SPECIAL_TOKENS = utils.retrieve_chat_template(
+        train_args.chat_tmpl_path
+    )
+    tokenizer = tokenizer_utils.setup_tokenizer(
+        train_args.model_path, SPECIAL_TOKENS, CHAT_TEMPLATE
+    )
+    dataset = token_dataset.setup_dataset(
+        os.path.join(train_args.data_output_dir, "data.jsonl"),
+    )
+    packing_max_batch_len, _ = (
+        multipack_sampler.find_packing_max_batch_len_and_grad_accum(
+            num_gpus=1,
+            avg_sample_len=dataset.get_lengths().mean(),
+            effective_batch_size=train_args.effective_batch_size,
+            max_batch_len_per_gpu=train_args.max_batch_len,
+            is_padding=train_args.is_padding_free,
+            pad_id=tokenizer.pad_token_id,
+            dataset=dataset,
+            seed=47,
+        )
+    )
+
+    collate_fn = partial(pad_collate_fn, pad_token_id=tokenizer.pad_token_id)
+
+    sampler = multipack_sampler.MultipackDistributedBatchSampler(
+        batch_max_length=packing_max_batch_len,
+        lengths=dataset.get_lengths(),
+        num_replicas=1,
+        rank=0,
+        seed=47,
+        padding=True,
+    )
+    sampler = {"batch_sampler": sampler}
+    dataloader = DataLoader(
+        dataset,
+        **sampler,
+        num_workers=4,
+        collate_fn=collate_fn,
+    )
+    dataloader.multiprocessing_context = "spawn"
+
+    dev = torch.device(train_args.device)
+    model = AutoModelForCausalLM.from_pretrained(train_args.model_path)
+    if len(tokenizer) > model.config.vocab_size:
+        print(
+            f"WARNING: tokenizer has {len(tokenizer)} tokens but model has {model.config.vocab_size} vocab size"
+        )
+        model.resize_token_embeddings(
+            int(8 * math.ceil(len(tokenizer) / 8.0))
+        )  # make the vocab size multiple of 8 for sharding the embedding layer.
+
+    # Fix any discrepancy between model and tokenizer
+    if (
+        model.config.pad_token_id is not None
+        and tokenizer.pad_token_id is not None
+        and model.config.pad_token_id != tokenizer.pad_token_id
+    ):
+        print(
+            f"WARNING: There is a mismatch between pad token id of model ({model.config.pad_token_id}) and tokenizer({tokenizer.pad_token_id}). Fixing model pad token id to be same as tokenizer's pad token id"
```

Comment:
Using logger warning here and update text
```suggestion
            f"There is a mismatch between pad token id of model ({model.config.pad_token_id}) and tokenizer ({tokenizer.pad_token_id}). Fixing model pad token id to be same as tokenizer's pad token id."
```